src/util/audio/AudioQueue.cpp

```diff
@@ -0,0 +1,62 @@
+#include "AudioQueue.h"
+
+AudioQueue::AudioQueue()
+{
+    XOJ_INIT_TYPE(AudioQueue);
+}
+
+AudioQueue::~AudioQueue()
+{
+    XOJ_CHECK_TYPE(AudioQueue);
+
+    XOJ_RELEASE_TYPE(AudioQueue);
+}
+
+void AudioQueue::reset()
+{
+    XOJ_CHECK_TYPE(AudioQueue);
+
+    this->notified = false;
+    this->streamEnd = false;
+    this->clear();
+}
+
+
+bool AudioQueue::empty()
+{
+    XOJ_CHECK_TYPE(AudioQueue);
+
+    return deque<int>::empty();
+}
+
+unsigned long AudioQueue::size()
+{
+    XOJ_CHECK_TYPE(AudioQueue);
+
+    return deque<int>::size();
+}
+
+void AudioQueue::push(int *samples, unsigned long nSamples)
+{
+    XOJ_CHECK_TYPE(AudioQueue);
+
+    unsigned long requiredSize = this->size() + nSamples;
+    if (this->max_size() < requiredSize)
```

Comment:
We use to have brace everywhere, because it causes less bugs if you do this consequent... (but this is not a blocker, and I write this only once)